### ISSUE TITLE
wishlist: explore snap socket activation support

### ISSUE THREAD

**Issue by tianon** [docker-snap/docker-snap#2]
I've been reading through https://forum.snapcraft.io/t/socket-activation-support/2050 + https://bugs.launchpad.net/snapd/+bug/1612440, and I'm thinking we should probably consider doing something there to match how Docker is deployed via APT.

My initial concern is making sure that our service is still "enabled" so that containers with a restart policy can start properly when the system reboots without the user needing to invoke some Docker command first. :smile:

(I'm also not sure how/if this would interact with https://github.com/snapcore/core20/issues/72 :sweat_smile:)

**Comment by tianon** [comment]
From my research so far, I don't think we'll be able to set the `docker` group on the socket if we switch to activation, so IMO it's not worth risking.

**Comment by anonymouse64** [comment]
Yes, initially socket activation would mean that you would not be able to set the docker group on the socket since you would need systemd directives to allow setting what group the socket file should be owned by, which is dependent on sorting out the extrausers stuff for the docker snap on classic which is a bit difficult to sort out. If/when we get there I will update this bug with the plan and how docker could use that.
